docs/emr_deployment.md

```diff
@@ -0,0 +1,126 @@
+---
+id: emr_deployment
+title: GeoTrellis EMR deployment with GDAL
+sidebar_label: GeoTrellis EMR deployment with GDAL
+---
+
+GeoTrellis can be used with _EMR 6_. However, if there is a need in GDAL bindings usage there can be some certain complexities.
+This section describes the entire _EMR 6_ deployment and GDAL installation in a way that would satisfy GeoTrellis GDAL requirements.
+
+Starting _GeoTrellis_ `3.4.2`, _GeoTrellis_ is compatible only with the GDAL`3.1 +` versions.
+
+## Installing GDAL on EMR
+
+The EMR installation conatans of the following steps:
+
+1. Install _GDAL_
+2. Set Spark `LD_LIBRARY_PATH`. In this case `GeoTrellis GDAL` would be able to use the installed `GDAL`.
+
+You can also check out [GeoTrellis Spark template](https://github.com/geotrellis/geotrellis-spark-job.g8) that is properly configured for the EMR jobs.
+
+### EMR Bootsrap script (GDAL installation on EMR)
+
+The EMR bootstrap script with all appropriate settings is located under the following S3 URI: [s3://geotrellis-test/emr-gdal/bootstrap.sh](s3://geotrellis-test/emr-gdal/bootstrap.sh).
+It installs GDAL 3.1.2 (by default) through conda and sets all appropriate env variables in case 
+it would be required to log in onto nodes and experiment with the `spark-shell`.
+
+```bash
+#!/bin/bash
+
+set -ex
+
+# The default GDAL version would be 3.1.2
+GDAL_VERSION=$1
+GDAL_VERSION=${GDAL_VERSION:="3.1.2"}
+
+# Install Conda
+wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh
+sudo sh Miniconda-latest-Linux-x86_64.sh -b -p /usr/local/miniconda
+
+source ~/.bashrc
+export PATH=/usr/local/miniconda/bin:$PATH
+
+# Install GDAL
+conda config --add channels conda-forge
+sudo pip3 install tqdm && \
+sudo /usr/local/miniconda/bin/conda install python=3.6 -y && \
+sudo /usr/local/miniconda/bin/conda install -c anaconda hdf5 -y && \
+sudo /usr/local/miniconda/bin/conda install -c conda-forge libnetcdf gdal=${GDAL_VERSION} -y
+
+echo "export PATH=/usr/local/miniconda/bin:$PATH" >> ~/.bashrc
+echo "export LD_LIBRARY_PATH=/usr/local/miniconda/lib/:/usr/local/lib:/usr/lib/hadoop/lib/native:/usr/lib/hadoop-lzo/lib/native:/docker/usr/lib/hadoop/lib/native:/docker/usr/lib/hadoop-lzo/lib/native:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib" >> ~/.bashrc
+```
+
+#### Spark settings (optional)
```

Comment:
It's unclear to me why this is optional or even ever required if the bootstrap script is already setting LD_LIBRARY_PATH. A sentence in this section that states _when_ and _why_ a user would want to set this bit of configuration would aid clarity a lot.

Reply:
@CloudNiner as you noticed bootstrap script is setting the `LD_LIBRARY_PATH` for the _shell_. 
EMR commands won't load it into the runtime.

Optional they are is due to the nature of the spark-submit. These spark settings set `spark-defaults`, and by defention these are optional default settings. You can set / override these `spark-defaults` during the command submission.

Reply:
I will add some clarifications, a good point :+1:

Reply:
> command submission

Shoot sorry I'm still a bit confused. Which command are you referring to? A `spark-submit` from an SSH shell session?

Reply:
@CloudNiner well in general I meant, that this LD_LIBRARY_PATH sets it for the ssh session conveniency, not for the EMR commands submission / for the spark-submissions with the yarn master.

Reply:
Gotcah so if I wanted to submit a job through the AWS EMR CLI I would need to provide the configurations in this optional section. 

The bootstrap script only sets the variables properly for the shell environment.

Reply:
@CloudNiner yep, this part is correct. But I called it optional is because instead of setting these spark-default setting you can pass them manually with the spark-submit command:

```sh
ingest-rastersource-ned-gdal:
	aws emr add-steps --output text --cluster-id ${CLUSTER_ID} \
--steps Type=CUSTOM_JAR,Name="Ingest RasterSource",Jar=command-runner.jar,Args=[\
spark-submit,--master,yarn-cluster,\
--class,geotrellis.contrib.performance.IngestRasterSource,\
--driver-memory,4200M,\
--driver-cores,2,\
--executor-memory,1500M,\
--executor-cores,1,\
--num-executors,200,\
--conf,spark.yarn.executor.memoryOverhead=700,\
--conf,spark.yarn.driver.memoryOverhead=700,\
--conf,spark.yarn.appMasterEnv.LD_LIBRARY_PATH=/usr/local/miniconda/lib/:/usr/local/lib,\
--conf,spark.executorEnv.LD_LIBRARY_PATH=/usr/local/miniconda/lib/:/usr/local/lib,\
${S3_URI},ned,gdal\
] | cut -f2 | tee last-step-id.txt
```